### PR TITLE
feat: run eppo-multiplatform tests for in-flight changes

### PR DESCRIPTION
**Pull request by typotter**
🎟️ towards FF-3085

👯‍♂️ **Related PRs**
- [eppo-multiplatform](https://github.com/Eppo-exp/eppo-multiplatform/pull/73)
- #57
- #58
- #59

### Motivation
Changes are often made to the [sdk-test-data](https://github.com/Eppo-exp/sdk-test-data) repository to capture new behaviours, bugs and edge cases. When these changes are pushed to `main`, the SDKs are cloned locally (locally to the github action running) and their respective tests are run. These tests are set up and run by copies of the SDK test workflows - see [sdk-test-data workflow](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows/test-sdks.yml). There are a number of limitations to this setup:

- Test steps are copied from the SDK’s respective workflows; changes to the SDK test workflows need to be replicated in sdk-test-data and this is not obvious to devs
- The SDK tests are not able to run against in-flight changes to `sdk-test-data`
- When new test-data is committed that breaks an SDK, that breakage is not surfaced in the SDK’s repository until some action triggers its testing workflow (on demand, on pull-request, etc.).

### Description of Changes
- Refactor testing for multiplatform SDK

External to this Change
🚀 - Each SDK's testing workflow is enhanced into a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), exposing parameters for SDK branch and the sdk-test-data branch to use in testing.

### CODE REVIEW

File: ufc/tests/test-case-integer-flag.json [hunk 1]
```diff
@@ -194,7 +194,7 @@
     {
       "subjectKey": "2",
       "subjectAttributes": {},
-      "assignment": 2,
+      "assignment": 99,
```
Comment (leoromanovsky) [inline ufc/tests/test-case-integer-flag.json:197]: I assume this was done to see if the tests would fail and I see in the github actions log that they do.

Comment (typotter) [reply ufc/tests/test-case-integer-flag.json]: yes- reverted

**Review by leoromanovsky — APPROVED** [review]
